service/collector/set.go

```diff
@@ -87,6 +100,10 @@ func NewSet(config SetConfig) (*Set, error) {
 			Logger: config.Logger,
 		}
 
+		if config.TrustedAdvisorEnabled {
+			c.Collectors = append(c.Collectors, trustedAdvisorCollector)
+		}
```

Comment:
We only run it when configured.

Reply:
How does this look like with BYOC? If a BYOC account can support trusted advisor, we should enable it for all accounts?

Reply:
I am not sure. How should this work? I think you are more into this topic. Is it that we then cannot have such a flag for the operator and have to dynamically detect that? Should that maybe be adapted later? I think I replicated what we have right now.

Reply:
Leaning towards dynamic detection, and dumping the flag. I can take it as part of https://github.com/giantswarm/giantswarm/issues/3609, this PR already helps in that direction. Sounds good.

Reply:
Should we merge this so I can get it done on my list and you take the work being left later when you are ready?